### DISCUSSION TITLE
Obsolete / unused classes?

### DISCUSSION THREAD

**Discussion by Ruud68** [Kunena/Kunena-Forum#9631] [General]
Hi,
I found the following (possibly not complete) list of unused (administrator) classes, can anybody tell me if these can be removed, or are there for future development?

- BadwordsController/Model/View
- BlockipsController/Model/View
- CloseControler/Model/View
- EmailControler/Model/View
- IconsControler/Model/View
- LabelsControler/Model/View
- StatsControler/Model/View

**Comment by xillibit** [comment]
Hello,

Most of them are obsolete and/or not used, i think you can keep the StatsControler/Model/View just in case

**Comment by Ruud68** [comment]
The StatsController is 'empty', I think this is confused with the statisticscontroller?
I will do a PR and set them as deprecated to be removed in Kunena version 7. That way we at least have some more guidance.

**Comment by Ruud68** [comment]
Ok, so did a PR for this. I have not set the stats to deprecated as I see now that this is a view (not working) that will show the forum stats (like they are displayed on the front-end), so not to be confused with the statistics view.

The stats view need some (?) work though before this can be 'enabled' as currently it is not finished and throwing a lot of errors
